### DISCUSSION TITLE
Best way to selectively prevent span recording?

### DISCUSSION THREAD

**Discussion by treyhyde** [open-telemetry/opentelemetry-ruby#640] [Q&A] [answered]
I'd like to cut down on the noise and data volume somewhat by excluding certain controller from tracing.   I looked through the rack instrumentation and threw in 

```
 OpenTelemetry::SDK.configure do |c|
    OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.config['untraced_endpoints'] = [
      'probes/ready'
    ]
 ...
```

to no effect.  Any best practices or suggestions for this case?  Thanks

**Comment by ericmustin** [comment]
@treyhyde That should do the trick, what's the rack `env['PATH_INFO']` for that endpoint? It would have to match that exactly 
https://github.com/open-telemetry/opentelemetry-ruby/blob/02fa36b7c2545997e4d8ad66ee382362eb6c2660/instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb#L89.

**Comment by fbogsany** [comment, marked as answer]
```ruby
OpenTelemetry::SDK.configure do |c|
  c.use 'OpenTelemetry::Instrumentation::Rack', { untraced_endpoints: ['probes/ready'] }
end
```
or
```ruby
OpenTelemetry::SDK.configure do |c|
  c.use_all { 'OpenTelemetry::Instrumentation::Rack' => { untraced_endpoints: ['probes/ready'] } }
end
```

Two things here:
1. `c.use` or `c.use_all` should be used to set config options for instrumentation rather than setting it directly with `OpenTelemetry::Instrumentation::Rack::Instrumentation.instance.config` - the latter will either not install the instrumentation correctly or will clobber its config.
2. Config option names are symbols rather than strings.

**Reply by treyhyde** [reply]
thank you!
